dataland-frontend/src/components/general/SlideShow.vue

```diff
@@ -0,0 +1,127 @@
+<template>
+  <div
+    ref="slider"
+    role="list"
+    :class="slidesContainerClasses"
+    @pointerdown="dragStartCondition"
+    @touchstart="dragStartCondition"
+  >
+    <slot />
+  </div>
+  <div v-if="slideCount > 1" :class="arrowsContainerClasses">
+    <button @click="move(-1)" aria-label="Previous slide" :class="leftArrowClasses" />
+    <button @click="move(1)" aria-label="Next slide" :class="rightArrowClasses" />
+  </div>
+</template>
+
+<script setup lang="ts">
+import { onUnmounted, ref, defineEmits, toRefs } from "vue";
+
+const props = withDefaults(
+  defineProps<{
+    slidesContainerClasses: string;
+    arrowsContainerClasses: string;
+    leftArrowClasses: string;
+    rightArrowClasses: string;
+    slideCount: number;
+    initialCenterSlide?: number;
+    scrollScreenWidthLimit: number;
+    slideWidth: number;
+  }>(),
+  {
+    initialCenterSlide: 0,
+    slideWidth: 440,
+  },
+);
+
+const { slideCount, initialCenterSlide, scrollScreenWidthLimit, slideWidth } = toRefs(props);
+
+const slider = ref<HTMLElement | null>(null);
```

Comment:
Again asking out of curiousity because this syntax is new to me:

Why can't we just write `ref<HTMLElement>()` ?   Because in this case, we are absoluetly 100% sure that the "div" that this is referenced to is actually exisiting as soon as this component is mounted.

Reply:
Using `ref<HTMLElement | null>(null)` explicitly states that the reference can be `null`, which is a more accurate representation of the lifecycle of a Vue component. When the component is not yet mounted or after it has been destroyed, the `ref` would be `null`.

Although you may be confident that the div exists as soon as the component is mounted, Vue itself has different phases in its lifecycle where the DOM element may not be available. Setting the type to `HTMLElement | null ` helps to account for these situations and makes it explicit that the reference may be empty at certain times, providing better type safety.

This encourages developers to check for `null` before using the ref, which is a good practice to avoid null reference errors.

Here's the Vue lifecycle for reference https://vuejs.org/guide/essentials/lifecycle.html